### DISCUSSION TITLE
Everyday Usage

### DISCUSSION THREAD

**Discussion by fdutey** [Homebrew/discussions#4642] [Everyday usage]
### Output of `brew config`

```shell
brew install wget => install 1 billion dependencies for other packages (ffmpeg, ruby, python ...)
```

### Output of `brew doctor`

```shell
...
```

### Description of issue

i just want to install wget to work, not install the whole universe on my machine

**Comment by ZhongRuoyu** [comment]
[Why does `brew upgrade <formula>` or `brew install <formula>` also upgrade a bunch of other stuff?](https://docs.brew.sh/FAQ#why-does-brew-upgrade-formula-or-brew-install-formula-also-upgrade-a-bunch-of-other-stuff)

> Homebrew doesn’t support arbitrary mixing and matching of formula versions, so everything a formula depends on, and everything that depends on it in turn, needs to be upgraded to the latest version as that’s the only combination of formulae we test. As a consequence any given `upgrade` or `install` command can upgrade many other (seemingly unrelated) formulae, especially if something important like `python` or `openssl` also needed an upgrade.
